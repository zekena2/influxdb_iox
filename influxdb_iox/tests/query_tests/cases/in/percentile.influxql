-- Query tests for influxql percentile
-- IOX_SETUP: percentile

--
-- Selectors
--
SELECT PERCENTILE(usage_idle,50) FROM cpu;
SELECT cpu,PERCENTILE(usage_idle,66.667),usage_system FROM cpu;
SELECT PERCENTILE(usage_idle,33.333) FROM cpu GROUP BY cpu;
SELECT PERCENTILE(usage_idle,90),usage_user FROM cpu WHERE cpu='3';
-- 0th percentile doesn't return any rows.
SELECT PERCENTILE(usage_idle,0) FROM cpu;

--
-- Aggregators
--
SELECT PERCENTILE(usage_system, 50), PERCENTILE(usage_system, 90), PERCENTILE(usage_system, 99) FROM cpu;
SELECT PERCENTILE(usage_system, 50), PERCENTILE(usage_system, 90), PERCENTILE(usage_system, 99) FROM cpu GROUP BY cpu;
SELECT PERCENTILE(usage_system, 50), PERCENTILE(usage_system, 90), PERCENTILE(usage_system, 99) FROM cpu WHERE time >= '1970-01-01 00:00:00' AND time < '1970-01-01 03:00:00'  GROUP BY time(1h),cpu;
