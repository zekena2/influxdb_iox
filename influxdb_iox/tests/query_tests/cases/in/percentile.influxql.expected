-- Test Setup: percentile
-- InfluxQL: SELECT PERCENTILE(usage_idle,50) FROM cpu;
name: cpu
+---------------------+------------+
| time                | percentile |
+---------------------+------------+
| 1970-01-01T01:10:09 | 49.7047    |
+---------------------+------------+
-- InfluxQL: SELECT cpu,PERCENTILE(usage_idle,66.667),usage_system FROM cpu;
name: cpu
+---------------------+-----+------------+--------------+
| time                | cpu | percentile | usage_system |
+---------------------+-----+------------+--------------+
| 1970-01-01T01:39:15 | 0   | 66.1469    | 99.8854      |
+---------------------+-----+------------+--------------+
-- InfluxQL: SELECT PERCENTILE(usage_idle,33.333) FROM cpu GROUP BY cpu;
name: cpu
tags: cpu=0
+---------------------+------------+
| time                | percentile |
+---------------------+------------+
| 1970-01-01T01:15:15 | 32.9757    |
+---------------------+------------+
name: cpu
tags: cpu=1
+---------------------+------------+
| time                | percentile |
+---------------------+------------+
| 1970-01-01T02:13:36 | 32.3062    |
+---------------------+------------+
name: cpu
tags: cpu=2
+---------------------+------------+
| time                | percentile |
+---------------------+------------+
| 1970-01-01T01:24:22 | 35.0742    |
+---------------------+------------+
name: cpu
tags: cpu=3
+---------------------+------------+
| time                | percentile |
+---------------------+------------+
| 1970-01-01T01:16:58 | 32.0821    |
+---------------------+------------+
name: cpu
tags: cpu=4
+---------------------+------------+
| time                | percentile |
+---------------------+------------+
| 1970-01-01T00:42:34 | 32.9685    |
+---------------------+------------+
-- InfluxQL: SELECT PERCENTILE(usage_idle,90),usage_user FROM cpu WHERE cpu='3';
name: cpu
+---------------------+------------+------------+
| time                | percentile | usage_user |
+---------------------+------------+------------+
| 1970-01-01T00:19:23 | 89.7011    | 34.7815    |
+---------------------+------------+------------+
-- InfluxQL: SELECT PERCENTILE(usage_idle,0) FROM cpu;
+------+------------+
| time | percentile |
+------+------------+
+------+------------+
-- InfluxQL: SELECT PERCENTILE(usage_system, 50), PERCENTILE(usage_system, 90), PERCENTILE(usage_system, 99) FROM cpu;
name: cpu
+---------------------+------------+--------------+--------------+
| time                | percentile | percentile_1 | percentile_2 |
+---------------------+------------+--------------+--------------+
| 1970-01-01T00:00:00 | 49.2732    | 89.754       | 99.0822      |
+---------------------+------------+--------------+--------------+
-- InfluxQL: SELECT PERCENTILE(usage_system, 50), PERCENTILE(usage_system, 90), PERCENTILE(usage_system, 99) FROM cpu GROUP BY cpu;
name: cpu
tags: cpu=0
+---------------------+------------+--------------+--------------+
| time                | percentile | percentile_1 | percentile_2 |
+---------------------+------------+--------------+--------------+
| 1970-01-01T00:00:00 | 49.7946    | 90.0001      | 98.8816      |
+---------------------+------------+--------------+--------------+
name: cpu
tags: cpu=1
+---------------------+------------+--------------+--------------+
| time                | percentile | percentile_1 | percentile_2 |
+---------------------+------------+--------------+--------------+
| 1970-01-01T00:00:00 | 50.148     | 89.4109      | 98.8158      |
+---------------------+------------+--------------+--------------+
name: cpu
tags: cpu=2
+---------------------+------------+--------------+--------------+
| time                | percentile | percentile_1 | percentile_2 |
+---------------------+------------+--------------+--------------+
| 1970-01-01T00:00:00 | 49.0258    | 89.7425      | 99.2486      |
+---------------------+------------+--------------+--------------+
name: cpu
tags: cpu=3
+---------------------+------------+--------------+--------------+
| time                | percentile | percentile_1 | percentile_2 |
+---------------------+------------+--------------+--------------+
| 1970-01-01T00:00:00 | 49.2054    | 89.9907      | 99.244       |
+---------------------+------------+--------------+--------------+
name: cpu
tags: cpu=4
+---------------------+------------+--------------+--------------+
| time                | percentile | percentile_1 | percentile_2 |
+---------------------+------------+--------------+--------------+
| 1970-01-01T00:00:00 | 48.1551    | 89.1691      | 98.9134      |
+---------------------+------------+--------------+--------------+
-- InfluxQL: SELECT PERCENTILE(usage_system, 50), PERCENTILE(usage_system, 90), PERCENTILE(usage_system, 99) FROM cpu WHERE time >= '1970-01-01 00:00:00' AND time < '1970-01-01 03:00:00'  GROUP BY time(1h),cpu;
name: cpu
tags: cpu=0
+---------------------+------------+--------------+--------------+
| time                | percentile | percentile_1 | percentile_2 |
+---------------------+------------+--------------+--------------+
| 1970-01-01T00:00:00 | 49.9884    | 89.7541      | 99.1413      |
| 1970-01-01T01:00:00 | 47.7725    | 90.8035      | 98.8471      |
| 1970-01-01T02:00:00 | 53.5363    | 90.0001      | 98.444       |
+---------------------+------------+--------------+--------------+
name: cpu
tags: cpu=1
+---------------------+------------+--------------+--------------+
| time                | percentile | percentile_1 | percentile_2 |
+---------------------+------------+--------------+--------------+
| 1970-01-01T00:00:00 | 48.2785    | 88.3004      | 98.7959      |
| 1970-01-01T01:00:00 | 51.1512    | 92.2132      | 98.9797      |
| 1970-01-01T02:00:00 | 49.6265    | 87.8342      | 98.0481      |
+---------------------+------------+--------------+--------------+
name: cpu
tags: cpu=2
+---------------------+------------+--------------+--------------+
| time                | percentile | percentile_1 | percentile_2 |
+---------------------+------------+--------------+--------------+
| 1970-01-01T00:00:00 | 50.0065    | 89.5125      | 99.109       |
| 1970-01-01T01:00:00 | 47.9867    | 89.5532      | 99.4226      |
| 1970-01-01T02:00:00 | 49.4459    | 90.439       | 99.2486      |
+---------------------+------------+--------------+--------------+
name: cpu
tags: cpu=3
+---------------------+------------+--------------+--------------+
| time                | percentile | percentile_1 | percentile_2 |
+---------------------+------------+--------------+--------------+
| 1970-01-01T00:00:00 | 46.7256    | 90.7002      | 99.3269      |
| 1970-01-01T01:00:00 | 50.7717    | 89.2459      | 98.9579      |
| 1970-01-01T02:00:00 | 49.6766    | 89.555       | 98.9499      |
+---------------------+------------+--------------+--------------+
name: cpu
tags: cpu=4
+---------------------+------------+--------------+--------------+
| time                | percentile | percentile_1 | percentile_2 |
+---------------------+------------+--------------+--------------+
| 1970-01-01T00:00:00 | 47.403     | 89.0086      | 98.9134      |
| 1970-01-01T01:00:00 | 50.6295    | 89.1826      | 98.9091      |
| 1970-01-01T02:00:00 | 46.1348    | 89.2463      | 98.7592      |
+---------------------+------------+--------------+--------------+